eitprocessing/continuous_data/__init__.py

```diff
@@ -45,11 +45,14 @@ def __post_init__(self) -> None:
         self.lock("time")
 
     def __setattr__(self, attr: str, value: Any):  # noqa: ANN401
-        old_value = getattr(self, attr)
-        if isinstance(old_value, np.ndarray) and old_value.flags["WRITEABLE"] is False:
-            msg = f"Attribute '{attr}' is locked and can't be overwritten."
-            raise AttributeError(msg)
-        super().__setattr__(self, attr, value)
+        try:
+            old_value = getattr(self, attr)
+            if isinstance(old_value, np.ndarray) and old_value.flags["WRITEABLE"] is False:
+                msg = f"Attribute '{attr}' is locked and can't be overwritten."
+                raise ValueError(msg)
```

Comment:
Why is this a ValueError, and not an AttributeError? 

The description of AttributeError seems very appropriate for this case.

> Raised when an attribute reference (see [Attribute references](dfile:///Users/peter/Library/Application%20Support/Dash/DocSets/Python_3/Python%203.docset/Contents/Resources/Documents/doc/reference/expressions.html#attribute-references)) or assignment fails.

I added a suggestion in the branch [slicedcopy_continuousdata_suggestions_psomhorst](https://github.com/EIT-ALIVE/eitprocessing/tree/slicedcopy_continuousdata_suggestions_psomhorst).

Reply:
We did it this way, because numpy normally also raises a ValueError when trying to change a non-WRITABLE array and figured it would make sense to stay close to that (plus it was simpler to code).
I do agree with you though, so will change it as you suggest.